File: 0204-count-primes/0204-count-primes.java
class Solution {
    public int countPrimes(int n) {
        if(n<=2){
            return 0;
        }
        
        boolean isPrimes[] = new boolean[n+1];
        isPrimes[n] = true;
        int m = n/2;

        for(int i=2; i<=m; i++){
            if(!isPrimes[i]){
                int j = n/i;
                for(int k=2; k<=j; k++){
                    isPrimes[i*k] = true;
                }
            }
        }

        int cnt = 0;
        for(int i=2; i<isPrimes.length; i++){
            if(!isPrimes[i]){
                cnt++;
            }
        }

        return cnt;
        
    }
}

File: 0204-count-primes/0204-count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if(n<=2):
            return 0
        
        isPrime = [False]*(n+1)
        isPrime[n] = True
        m = n//2

        for i in range(2, m+1):
            j = n//i
            for k in range(2, j+1):
                isPrime[i*k] = True
        
        cnt = 0
        for i in range(2, len(isPrime)):
            if(not(isPrime[i])):
                cnt+=1
        
        return cnt
        